### PR TITLE
Fix base color and reflectivity calculations for PBR materials

### PR DESCRIPTION
**Pull request by jjhembd**
# Description

Our calculation of the base diffuse color for a material was scaled by the transmission coefficient `(1.0 - f0)` (where `f0` is the specular reflection coefficient at normal incidence):
```glsl
material.diffuse = material.diffuse * (1.0 - f0) * (1.0 - metalness);
```
This intuitively seems to make sense at normal incidence. The incident light that actually interacts with the microfacets of the surface (in the diffuse model) will be reduced by the amount of light already reflected specularly.

However, this simple 1D scaling is an oversimplification of something that is already accounted for in the BRDF, so including the scaling results in PBR materials being rendered slightly too dim.

This PR instead computes the base diffuse color using a formula more similar to the [glTF Sample Viewer](https://github.com/KhronosGroup/glTF-Sample-Viewer):
```glsl
material.diffuse = mix(material.baseColor.rgb, vec3(0), metalness);
```

Note the new `baseColor` property on the `czm_modelMaterial` struct. This helps avoid confusion from overwriting the `diffuse` property too many times, and fixes https://github.com/CesiumGS/cesium/issues/12041, where the `KHR_materials_specular` extension was using the `diffuse` property as if it were the raw base color.

## Issue number and link

Resolves https://github.com/CesiumGS/cesium/issues/12041

## Testing plan

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- ~[ ] I have added or updated unit tests to ensure consistent code coverage~
- ~[ ] I have update the inline documentation, and included code examples where relevant~
- [x] I have performed a self-review of my code

### CODE REVIEW

File: packages/engine/Source/Shaders/Model/MaterialStageFS.glsl [hunk 1]
```diff
@@ -296,12 +296,12 @@ float setMetallicRoughness(inout czm_modelMaterial material)
 
     // dielectrics use f0 = 0.04, metals use albedo as f0
     const vec3 REFLECTANCE_DIELECTRIC = vec3(0.04);
-    vec3 f0 = mix(REFLECTANCE_DIELECTRIC, material.diffuse, metalness);
+    vec3 f0 = mix(REFLECTANCE_DIELECTRIC, material.baseColor.rgb, metalness);
 
     material.specular = f0;
 
     // diffuse only applies to dielectrics.
-    material.diffuse = material.diffuse * (1.0 - f0) * (1.0 - metalness);
```
Comment (jjhembd) [inline packages/engine/Source/Shaders/Model/MaterialStageFS.glsl:304]: I have not been able to trace why we multiplied by `(1.0 - f0)` here.

File: CHANGES.md [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 ##### Fixes :wrench:
 
+- Fixed diffuse color calculation for PBR materials. Many models will now appear slightly brighter. This also fixes an [issue affecting the KHR_materials_specular extension](https://github.com/CesiumGS/cesium/issues/12041). [#12043](https://github.com/CesiumGS/cesium/pull/12043)
```
Comment (ggetz) [inline CHANGES.md:17]: Even though they were fixed in one PR, I would suggest listing this as two different items. The former (`Fixed diffuse color calculation for PBR materials`) can link to the PR. The later (`fixes an issue affecting the KHR_materials_specular extension`) can link to the issue.

File: packages/engine/Source/Shaders/Model/MaterialStageFS.glsl [hunk 3]
```diff
@@ -296,12 +296,12 @@ float setMetallicRoughness(inout czm_modelMaterial material)
...
+    material.diffuse = mix(material.baseColor.rgb, vec3(0), metalness);
```
Comment (ggetz) [inline packages/engine/Source/Shaders/Model/MaterialStageFS.glsl:304]: Shouldn't this be?

```suggestion
    material.diffuse = mix(material.baseColor.rgb, vec3(0.0), metalness);
```

**Comment by jjhembd** [comment]
Thanks @ggetz! I addressed both of those comments.

**Comment by ggetz** [comment]
All looks good! Thanks @jjhembd!

**Comment by jjhembd** [comment]
For the record: the extra `(1 - f0)` factor has appeared in other implementations as well. See https://github.com/KhronosGroup/glTF/issues/1945 and https://github.com/KhronosGroup/glTF/pull/2022 for discussion about how it was caught and corrected in the glTF spec implementation notes.

**Pull request merged**